win32ss/drivers/videoprt/registry.c

```diff
@@ -586,6 +586,10 @@ IntCreateNewRegistryPath(
         return Status;
     }
 
+    /* Copy the saved resolution to the source key */
+    IntCopyRegistryValue(NewKey, SettingsKey, L"DefaultSettings.XResolution");
+    IntCopyRegistryValue(NewKey, SettingsKey, L"DefaultSettings.YResolution");
```

Comment:
Then, at line 581, you would "need" to add `KEY_SET_VALUE` or `KEY_WRITE` or the like.

Reply:
No changes are needed there. Please see my comments above.

Reply:
Probably you want to check here the Status of both IntCopyRegistryValue calls. If in some emulators resolution is still failing is because probably one (or both) of these calls are failing and it is going unnoticed.

Reply:
> No changes are needed there.

That seems rather inconsistent to me: asking for read only, then doing a write.

Reply:
@vicmarcal, You may be correct, but if you look at Line-350 and Line-351 above you will see similar coding by @tkreuzer and there is no testing for success there. I normally like to follow the lead of previous developers unless I know of a good reason not to do so. Let's see what others think here.
I do have some ideas of how to trouble-shoot the other problems and that would be to add the following two lines just above the current new ones:
```
    DPRINT("SettingsKey is '%wZ'.\n", &DeviceExtension->RegistryPath);
    DPRINT("NewKey is '%wZ'.\n", &DeviceExtension->NewRegistryPath);
```
These could be "DPRINT1" as well. Maybe we might want to include these now. Let's see what others think.
 Thanks.